api/v1/constructors.go

```diff
@@ -1437,30 +1437,10 @@ func NewPlatformNetworkSpec(pool addresspools.AddressPool, network networks.Netw
 		allocation_type = networks.AllocationOrderStatic
 	}
 	spec := PlatformNetworkSpec{
-		Type:               network.Type,
-		Subnet:             pool.Network,
-		FloatingAddress:    pool.FloatingAddress,
-		Controller0Address: pool.Controller0Address,
-		Controller1Address: pool.Controller1Address,
-		Prefix:             pool.Prefix,
-		Gateway:            pool.Gateway,
-		Allocation: AllocationInfo{
-			Type:  allocation_type,
-			Order: &pool.Order,
-		},
+		NetworkType:      network.Type,
```

Comment:
Just that it will be clearer what `type` we are talking about.
I think we could still go ahead with `type` instead of `networkType` now that PlatformNetwork resource doesn't mix up AddressPool, what do you suggest @yjian118 ?